### PR TITLE
Add GPIO support for i.MX8MP

### PR DESCRIPTION
**Pull request by Siviuze**
## Summary
 * Add GPIO support with second level IRQ
 * Add userleds for Verdin board
 * Add buttons for Verdin board

### CODE REVIEW

File: boards/arm/mx8mp/verdin-mx8mp/configs/nsh/defconfig [hunk 1]
```diff
@@ -5,23 +5,43 @@
...
 CONFIG_ARCH_INTERRUPTSTACK=2048
+CONFIG_ARCH_IRQBUTTONS=y
 CONFIG_ARCH_STACKDUMP=y
 CONFIG_ARMV7M_DCACHE=y
 CONFIG_ARMV7M_DCACHE_WRITETHROUGH=y
 CONFIG_ARMV7M_ICACHE=y
 CONFIG_ARM_MPU=y
 CONFIG_BOARD_LOOPSPERMSEC=159972
 CONFIG_BUILTIN=y
+CONFIG_CXX_EXCEPTION=y
```
Comment (trns1997) [inline boards/arm/mx8mp/verdin-mx8mp/configs/nsh/defconfig:26]: do you really need cxx stuff by default?

File: boards/arm/mx8mp/verdin-mx8mp/src/Makefile [hunk 2]
```diff
@@ -34,5 +34,15 @@ ifeq ($(CONFIG_MX8MP_I2C_DRIVER),y)
   CSRCS += mx8mp_i2cdev.c
 endif
 
+ifeq ($(CONFIG_ARCH_LEDS),y)
+  #CSRCS += mx8mp_autoleds.c
```
Comment (trns1997) [inline boards/arm/mx8mp/verdin-mx8mp/src/Makefile:38]: commented code

File: boards/arm/mx8mp/verdin-mx8mp/src/mx8mp_userleds.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: led_dumppins
+ ****************************************************************************/
+
+#ifdef LED_VERBOSE
+static void led_dumppins(const char *msg)
+{
+  nrf53_pin_dump(PINCONFIG_LED, msg);
```
Comment (trns1997) [inline boards/arm/mx8mp/verdin-mx8mp/src/mx8mp_userleds.c:55]: incorrect board

File: boards/arm/mx8mp/verdin-mx8mp/configs/nsh/defconfig (hunk 1 above)
Comment (Siviuze) [reply boards/arm/mx8mp/verdin-mx8mp/configs/nsh/defconfig:26]: For my current usage yes :)

I may add another defconfig later though.

File: boards/arm/mx8mp/verdin-mx8mp/src/mx8mp_userleds.c (hunk 3 above)
Comment (Siviuze) [reply boards/arm/mx8mp/verdin-mx8mp/src/mx8mp_userleds.c]: Oh right! Fixed but no logic yet.

File: boards/arm/mx8mp/verdin-mx8mp/src/Makefile (hunk 2 above)
Comment (Siviuze) [reply boards/arm/mx8mp/verdin-mx8mp/src/Makefile]: Fixed but no logic yet.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**